### ISSUE TITLE
Local search only works in deployed scenario

### ISSUE THREAD

**Issue by JohnCampionJr** [vuejs/vitepress#2237]
### Describe the bug

In dev mode, the local search box throws an error in the browser console.

```
caught (in promise) SyntaxError: The requested module '/node_modules/.pnpm/mark.js@8.11.1/node_modules/mark.js/dist/mark.js?v=e66c0727' does not provide an export named 'default' (at VPLocalSearchBox.vue:12:8)
```

Works fine at build/deploy time.

Any way we can make this work in dev mode?

### Reproduction

Run a brand new vitepress site with the getting started, enable local search and click the search box

### Expected behavior

At least not throw an error, even better if there was some way to make it work.

### System Info

```shell
NA
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Yeah we are working on it. But not getting that error 👀. For us the index is blank.

**Comment by FlorianKoerner** [comment]
Also reproducible with vitepress.new.

**Comment by brc-dd** [comment]
Ah, thanks. It probably those interop issues. It won't just work even if we fix it.

**Comment by brc-dd** [comment]
Fixed that error in 1b0a249

**Comment by brc-dd** [comment]
Should work now, https://stackblitz.com/edit/vite-8a6qwm

<img width="1840" alt="image" src="https://user-images.githubusercontent.com/40380293/232335668-567be2b4-b077-4bc9-8270-074f20f0150f.png">

There are some rough edges, we are working on fixing them
